akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpMessageParser.scala

```diff
@@ -163,7 +164,14 @@ private[http] trait HttpMessageParser[Output >: MessageOutput <: ParserOutput] {
         case h: `Content-Type` => cth match {
           case None      => parseHeaderLines(input, lineEnd, headers, headerCount + 1, ch, clh, Some(h), teh, e100c, hh)
           case Some(`h`) => parseHeaderLines(input, lineEnd, headers, headerCount, ch, clh, cth, teh, e100c, hh)
-          case _         => failMessageStart("HTTP message must not contain more than one Content-Type header")
+          case Some(x) if isResponseParser =>
+            import ConflictingResponseContentTypeHeaderProcessingMode._
+            settings.conflictingResponseContentTypeHeaderProcessingMode match {
+              case Error => failMessageStart("HTTP message must not contain more than one Content-Type header")
+              case First => parseHeaderLines(input, lineEnd, headers += h, headerCount + 1, ch, clh, cth, teh, e100c, hh)
+              case Last  => parseHeaderLines(input, lineEnd, headers += x, headerCount + 1, ch, clh, Some(h), teh, e100c, hh)
```

Comment:
I like that the colliding headers are not discarded but transparently provided as plain headers.